dev-infrastructure/modules/aks-cluster-base.bicep

```diff
@@ -37,6 +37,31 @@ param podSubnetPrefix string
 param clusterType string
 param workloadIdentities array
 
+@description('IPTags to be set on the cluster outbound IP address in the format of ipTagType:tag,ipTagType:tag')
+param aksClusterOutboundIpAddressIPTags string = ''
+var aksClusterOutboundIpAddressIPTagsArray = [
+  for tag in (aksClusterOutboundIpAddressIPTags == '') ? [] : split(aksClusterOutboundIpAddressIPTags, ','): {
+    ipTagType: split(tag, ':')[0]
+    tag: split(tag, ':')[1]
+  }
+]
+
+@description('IPTags to be set on the Istio Ingress Gateway IP address in the format of ipTagType:tag,ipTagType:tag')
+param istioIngressGatewayIpAddressIPTags string = ''
+var istioIngressGatewayIpAddressIPTagsArray = [
+  for tag in (istioIngressGatewayIpAddressIPTags == '') ? [] : split(istioIngressGatewayIpAddressIPTags, ','): {
+    ipTagType: split(tag, ':')[0]
+    tag: split(tag, ':')[1]
+  }
+]
+
+@description('List of Availability Zones for zone-redundant resources')
+param zoneRedundancyZones array = [
```

Comment:
I think we will need to check the region to see if it supports zones 
https://learn.microsoft.com/en-us/azure/reliability/availability-zones-region-support 

Does it make sense to move that logic to the publicipaddress module?

Reply:
We currently do not do that check for the AKS cluster itself, so I figure this could be left out of this task for now to avoid scope creep

I did parametrize the zones array which is passed to the IP Address module and now used in the AKS cluster definition

Reply:
Ah, you're right. It makes sense to get that in a follow up.